### ISSUE TITLE
Discuss implementation of the timeline series

### ISSUE THREAD

**Issue by TorsteinHonsi** [highcharts/highcharts#9603]
Main concern is that the current proposal doesn't use a datetime axis. This should at least be an option. Another concern is the `point.date` option which is currently a string. The convention in Highcharts is to use either `x`, which would be a number (timestamp in case of a datetime axis), or `name`, which would be applied directly on the X axis in a category axis. 

As far as I can see, the timeline series should use datetime or category xAxis optionally, where the category axis would give the same result as the current implementation.

If the team agrees on this, we will postpone the official release of the timeline series until after v7.

**Comment by Denyllon** [comment]
Thank you for your opinion on this topic. I looked through many of timelines before start implementing it, and noticed that vast majority of them has the same time interval between points. Currently, the timeline calculates all events `x` values automatically (very similar to treemap), so user doesn't have to take care about saving the same time between them. It distributes whole available space between points equally.

![image](https://user-images.githubusercontent.com/15111955/49581055-0537a480-f951-11e8-8e2c-70bd8d992224.png)

I agree that the `point.date` could be the timestamp, and would be great enhancement, but it will make the timeline series implementation a bit more complicated than it's now.

Before starting rebuilding process, please make me sure about how exactly it should work, and which options should be given for user to set.

**Comment by TorsteinHonsi** [comment]
Thanks for your input! I think we can achieve both the equal-interval layout (like above) and a true timeline easily. But I think we should postpone this until v7.1. Here's my proposal.

* Get rid of the `point.date` option.
* For equal intervals, use `point.name` and `xAxis.type: 'category'`. This is how a category axis usually works. We get all the positioning and axis logic for free and it is consistent  with how things are usually done.
* For true datetime intervals, use `point.x` and `xAxis.type: 'datetime'`.
* Set up demos for both.
